pkg/factory/config.go

```diff
@@ -107,13 +109,30 @@ func (c *Configuration) validate() (bool, error) {
 
 type Sbi struct {
 	Scheme       string `yaml:"scheme" valid:"scheme"`
-	RegisterIPv4 string `yaml:"registerIPv4,omitempty" valid:"host,required"` // IP that is registered at NRF.
-	BindingIPv4  string `yaml:"bindingIPv4,omitempty" valid:"host,required"`  // IP used to run the server in the node.
-	Port         int    `yaml:"port,omitempty" valid:"port,required"`
+	RegisterIPv4 string `yaml:"registerIPv4,omitempty" valid:"host,optional"` // IP that is registered at NRF.
+	RegisterIPv6 string `yaml:"registerIPv6,omitempty" valid:"host,optional"` // IP that is registered at NRF.
+	BindingIPv4  string `yaml:"bindingIPv4,omitempty" valid:"host,optional"`  // IP used to run the server in the node.
+	BindingIPv6  string `yaml:"bindingIPv6,omitempty" valid:"host,optional"`  // IP used to run the server in the node.
+	Port         int    `yaml:"port,omitempty" valid:"port,required,with_register,with_binding"`
 	Tls          *Tls   `yaml:"tls,omitempty" valid:"optional"`
 }
 
 func (s *Sbi) validate() (bool, error) {
+	govalidator.CustomTypeTagMap.Set("with_register", func(i interface{}, context interface{}) bool {
+		switch v := context.(type) {
+		case Sbi:
+			return v.RegisterIPv4 != "" || v.RegisterIPv6 != ""
```

Comment:
great idea to add a check here!

we should also maybe check that the two cannot be set at the same time, or else, support the registration/binding at both IPv4 and IPv6 at the same time?

Reply:
As a first time, we can only check two cannot be set at the same time, and ask the advice of the Free5gc team, that sounds good for you?